### PR TITLE
fix(`forge`): detect invalid inline config keys

### LINKED ISSUE #5371: bug: inline config not catching some errors
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [ ] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (e488e2b 2023-07-10T15:17:42.605282000Z)

### What command(s) is the bug in?

forge test

### Operating System

None

### Describe the bug

Config file:
```toml
[profile.default]
# --- snip ---

[profile.ci]
# --- snip ---

[profile.lite]
# --- snip ---
```

Test
```solidity
/// forge-config: default.fuzz.runs = 1
/// forge-config: default.ci.runs = 1
/// forge-config: default.lite.runs = 1
function test1() public {}
```

This config is wrong: there is no `ci` or `lite` config key in the default profile, so I'd have expected this error to be caught. The correct inline config is:

```solidity
/// forge-config: default.fuzz.runs = 1
/// forge-config: ci.fuzz.runs = 1
/// forge-config: lite.fuzz.runs = 1
function test1() public {}
```

Other inline config errors are caught. For example, with `/// forge-config: default.fuzz.runssss = 1`, running `forge test` will error says `Inline config error detected at test/ERC5564Announcer.t.sol:Announce:test1`

So, two pieces of feedback:
1. Also catch when the first key is not valid
2. Improve the error message to write out the incorrect key so bugs are easier to find

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Closes #5371 

## Solution

- Introduce `const VALID_CONFIG_KEYS` 
- Parse the config lines and see if it contains any keys that are outside `VALID_CONFIG_KEYS`.
- Throw `InvalidConfigKey` error if it does.

### CODE REVIEW

File: crates/config/src/inline/conf_parser.rs [hunk 1]
```diff
@@ -41,10 +48,27 @@ where
...
+                    .nth(1) // Get the part after the first dot (profile.KEY.value)
+                    .unwrap_or("unknown-config-key")
+                    .to_string();
+                return Err(InlineConfigError {
+                    line,
+                    source: InlineConfigParserError::InvalidConfigKey(
+                        wrong_key,
+                        format!("{VALID_CONFIG_KEYS:?}"),
+                    ),
+                });
```
Comment (zerosnacks) [inline crates/config/src/inline/conf_parser.rs:65]: I think a warning would be sufficient here given that the key is simply ignored when continuing

File: crates/config/src/inline/conf_parser.rs [hunk 2]
```diff
@@ -2,6 +2,13 @@ use super::{remove_whitespaces, InlineConfigParserError};
 use crate::{inline::INLINE_CONFIG_PREFIX, InlineConfigError, NatSpec};
 use regex::Regex;
 
+/// Used to detect invalid configuration keys in the inline config.
+///
+/// # Example
+///
+/// forge-config: default.fuzz.runs = 100 - `fuzz` is a valid key
+/// forge-config: default.wrong.runs = 100 - `wrong` is an invalid key
+const VALID_CONFIG_KEYS: [&str; 2] = ["fuzz", "invariant"];
```
Comment (zerosnacks) [inline crates/config/src/inline/conf_parser.rs:11]: For now this is sufficient I think, assuming this functionality will be updated when we have full support for in-line configuration

Comment (mattsse) [inline crates/config/src/inline/conf_parser.rs:11]: I wonder if this isn't too simplistic?

Comment (yash-atreya) [reply crates/config/src/inline/conf_parser.rs:11]: It is for now. But it won't be needed when we have full support for inline config.

**Comment by yash-atreya** [comment]
@grandizzy can we merge this?

**Comment by yash-atreya** [comment]
Superseded by #9414

**Pull request closed without merging**